src/couch_mrview/src/couch_mrview_http.erl

```diff
@@ -486,8 +491,43 @@ parse_params(Props, Keys, #mrargs{} = Args0, Options) ->
                 Args0;
             _ ->
                 % group_level set to undefined to detect if explicitly set by user
-                Args0#mrargs{keys = Keys, group = undefined, group_level = undefined}
+                case Keys of
+                    [_] ->
+                        Args0#mrargs{group = undefined, group_level = undefined};
+                    _ ->
+                        Args0#mrargs{keys = Keys, group = undefined, group_level = undefined}
+                end
         end,
+
+    PropsKeys = proplists:get_keys(Props),
+    case config:get_boolean("chttpd", "enable_key_exclusivity", ?DEFAULT_ENABLE_KEY_EXCLUSIVITY) of
```

Comment:
A separate function would be clearer as nick suggests. we can have separate clauses for the `true` and `false` logic.

Reply:
I don't find the sets suggestion much tidier than the lists:foldl, how about;

```
lists:any(fun(K) -> lists:member(K, ["start_key", "startkey", "end_key", "endkey"]) end, PropsKeys).
```

?

Reply:
good idea, lists:any is more clear!

Reply:
Agree. I tried with `timer:tc()`, `lists:any()` is more efficient.
```erl
% first time
Any: {0,true}
Foldl: {1,true}
Sets: {1770,true}

% second time
Any: {0,true}
Foldl: {1,true}
Sets: {12,true}
```
Thanks for showing me another way to check intersections.